src/js/apps/globals/nav_app.js

```diff
@@ -222,4 +226,30 @@ export default App.extend({
 
     navView.triggerMethod('search:active', true);
   },
+  showAddPatientModal() {
+    const addPatientView = new AddPatientModal({
+      model: Radio.request('entities', 'patients:model', {}),
+    });
+
+    // addPatientView.setState(addPatientState);
+
+    const addPatientModal = Radio.request('modal', 'show:custom', addPatientView);
+
+    this.listenTo(addPatientModal, {
+      'save'({ model }) {
+        model.saveAll(model.attributes)
+          .then(({ data }) => {
+            Radio.trigger('event-router', 'patient:dashboard', data.id);
+            addPatientModal.destroy();
+          })
+          .fail(({ responseJSON }) => {
+            addPatientView.setState({
+              errors: {
+                name: responseJSON.errors[0].detail,
```

Comment:
Lets add a comment here that says that our mvp implementation assumes handling only the similar-patient server error.